File: src/main/java/mediathek/gui/tabs/tab_film/PsetButtonsPanel.java
package mediathek.gui.tabs.tab_film;

import mediathek.config.Daten;
import mediathek.gui.messages.ProgramSetChangedEvent;
import mediathek.tool.MessageBus;
import net.engio.mbassy.listener.Handler;
import org.jdesktop.swingx.WrapLayout;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;

public class PsetButtonsPanel extends JPanel {
    private static final int DEFAULT_HEIGHT = 90;
    private final JPanel btnPanel = new JPanel();

    private final GuiFilme guiFilme;

    public PsetButtonsPanel(@NotNull GuiFilme guiFilme) {
        this.guiFilme = guiFilme;

        setLayout(new BorderLayout());
        setPreferredSize(new Dimension(Integer.MAX_VALUE, DEFAULT_HEIGHT));
        setMinimumSize(new Dimension(100, DEFAULT_HEIGHT));

        btnPanel.setLayout(new WrapLayout(FlowLayout.LEFT, 5, 5));

        JScrollPane sp = new JScrollPane();
        add(sp, BorderLayout.CENTER);

        sp.setViewportView(btnPanel);

        setupButtonLayout();

        MessageBus.getMessageBus().subscribe(this);
    }

    public void install(@NotNull JTabbedPane tabbedPane) {
        tabbedPane.add("Buttons", this);
    }

    @Handler
    private void handleProgramSetChangedEvent(ProgramSetChangedEvent e) {
        System.out.println("Handle PSET CHANGE");
        SwingUtilities.invokeLater(this::setupButtonLayout);
    }

    protected void setupButtonLayout() {
        btnPanel.removeAll();

        for (var pset : Daten.listePset.getListeButton()) {
            if (!pset.isFreeLine()) {
                JComponent comp;

                if (pset.isLabel()) {
                    comp = new JLabel(pset.getName());
                } else {
                    var b = new JButton(pset.getName());
                    b.addActionListener(l -> guiFilme.playerStarten(pset));
                    comp = b;
                }

                pset.getForegroundColor().ifPresent(comp::setForeground);
                btnPanel.add(comp);
            } else {
                btnPanel.add(new JLabel(""));
            }
        }
        validate();
        repaint();
    }
}

File: src/main/kotlin/mediathek/gui/tabs/tab_film/PsetButtonsPanel.kt
package mediathek.gui.tabs.tab_film

import mediathek.daten.DatenPset
import mediathek.daten.ProgramSetRepository
import mediathek.gui.messages.ProgramSetChangedEvent
import mediathek.tool.MessageBus.messageBus
import net.engio.mbassy.listener.Handler
import org.jdesktop.swingx.WrapLayout
import java.awt.BorderLayout
import java.awt.Dimension
import java.awt.FlowLayout
import java.util.function.Consumer
import javax.swing.*

class PsetButtonsPanel(
    private val programSets: ProgramSetRepository,
    private val startFilmWithPset: Consumer<DatenPset>,
) : JPanel() {
    private val btnPanel = JPanel()

    init {
        layout = BorderLayout()
        preferredSize = Dimension(Int.MAX_VALUE, DEFAULT_HEIGHT)
        minimumSize = Dimension(100, DEFAULT_HEIGHT)

        btnPanel.layout = WrapLayout(FlowLayout.LEFT, 5, 5)

        val scrollPane = JScrollPane()
        add(scrollPane, BorderLayout.CENTER)
        scrollPane.viewport.view = btnPanel

        setupButtonLayout()

        messageBus.subscribe(this)
    }

    fun install(tabbedPane: JTabbedPane) {
        tabbedPane.add("Buttons", this)
    }

    @Suppress("UNUSED_PARAMETER")
    @Handler
    private fun handleProgramSetChangedEvent(event: ProgramSetChangedEvent) {
        SwingUtilities.invokeLater(this::setupButtonLayout)
    }

    protected fun setupButtonLayout() {
        btnPanel.removeAll()

        for (pset in programSets.list.listeButton) {
            if (!pset.isFreeLine) {
                val component: JComponent = if (pset.isLabel) {
                    JLabel(pset.name)
                } else {
                    JButton(pset.name).apply {
                        addActionListener { startFilmWithPset.accept(pset) }
                    }
                }

                pset.foregroundColor.ifPresent(component::setForeground)
                btnPanel.add(component)
            } else {
                btnPanel.add(JLabel(""))
            }
        }

        validate()
        repaint()
    }

    companion object {
        private const val DEFAULT_HEIGHT = 90
    }
}
